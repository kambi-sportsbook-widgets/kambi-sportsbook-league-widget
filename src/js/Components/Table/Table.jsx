import React, { PropTypes } from 'react';
import styles from './Table.scss';

const Table = ({ children }) => {
   return (
      <table className={['KambiWidget-card-background-color', styles.general].join(' ')}>
         {children}
      </table>
   );
};

Table.propTypes = {
   /**
    * Inner elements
    */
   children: PropTypes.node
};

export default Table;
