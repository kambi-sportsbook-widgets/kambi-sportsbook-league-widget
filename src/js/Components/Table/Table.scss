.general {
   border-spacing: 0;
   border-collapse: collapse;
   width: 100%;
}
