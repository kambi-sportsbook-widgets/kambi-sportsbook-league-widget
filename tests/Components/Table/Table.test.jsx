/* eslint-env jest */
import React, { Children } from 'react';
import Table from '../../../src/js/Components/Table/Table';
import ReactTestRenderer from 'react-test-renderer';

describe('Table DOM rendering', () => {

   it('renders correctly without children', () => {
      const tree = ReactTestRenderer.create(
         <Table />
      ).toJSON();

      expect(tree).toMatchSnapshot();
   });

   it('renders correctly with children', () => {
      const tree = ReactTestRenderer.create(
         <Table>
            <tr><td/><td/></tr>
            <tr><td/><td/></tr>
         </Table>
      ).toJSON();

      expect(tree).toMatchSnapshot();
   });

});
