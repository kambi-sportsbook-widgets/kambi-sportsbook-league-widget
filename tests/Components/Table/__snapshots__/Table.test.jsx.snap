// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table DOM rendering renders correctly with children 1`] = `
<table
  className="KambiWidget-card-background-color general"
>
  <tr>
    <td />
    <td />
  </tr>
  <tr>
    <td />
    <td />
  </tr>
</table>
`;

exports[`Table DOM rendering renders correctly without children 1`] = `
<table
  className="KambiWidget-card-background-color general"
/>
`;
